stonesoup/detector/tensorflow.py

```diff
@@ -0,0 +1,198 @@
+            width and height of the bounding box. Additionally, each detection
+            carries the following meta-data fields:
+
+            - ``raw_box``: The raw bounding box, as generated by TensorFlow.
+
+            - ``class``: A dict with keys ``id`` and ``name`` relating to the \
+            id and name of the detection class, as specified in the label map.
+
+            - ``score``: A float in the range ``(0, 1]`` indicating the \
+            detector's confidence
+        """
+
+        if self.run_async:
+            yield from self._detections_gen_async()
+        else:
+            yield from self._detections_gen()
+
+    def _capture(self):
+        for timestamp, frame in self.sensor:
+            self._thread_lock.acquire()
+            self._buffer = frame
+            self._thread_lock.release()
+
+    def _detections_gen(self):
+        for timestamp, frame in self.sensor:
+            detections = self._get_detections_from_frame(frame)
+            yield timestamp, detections
+
+    def _detections_gen_async(self):
+        while self._capture_thread.is_alive():
+            if self._buffer is not None:
+                self._thread_lock.acquire()
+                frame = copy(self._buffer)
+                self._buffer = None
+                self._thread_lock.release()
+
+                detections = self._get_detections_from_frame(frame)
+
+                yield frame.timestamp, detections
+
+    def _get_detections_from_frame(self, frame):
+        # Expand dimensions since the model expects images
+        # to have shape: [1, None, None, 3]
+        image_np_expanded = np.expand_dims(frame.pixels, axis=0)
+        # Extract image tensor
+        image_tensor = self._graph.get_tensor_by_name(
+            'image_tensor:0')
+        # Extract detection boxes
+        boxes = self._graph.get_tensor_by_name(
+            'detection_boxes:0')
+        # Extract detection scores
+        scores = self._graph.get_tensor_by_name(
+            'detection_scores:0')
+        # Extract detection classes
+        classes = self._graph.get_tensor_by_name(
+            'detection_classes:0')
+        # Extract number of detections
+        num_detections = self._graph.get_tensor_by_name(
+            'num_detections:0')
+
+        # Perform detection
+        (boxes, scores, classes, num_detections) = self._sess.run(
+            [boxes, scores, classes, num_detections],
+            feed_dict={image_tensor: image_np_expanded})
+
+        classes = np.squeeze(classes).astype(np.int32)
+        boxes = np.squeeze(boxes)
+        scores = np.squeeze(scores)
+
+        # Empty detection set
+        frame_height, frame_width, _ = frame.pixels.shape
+        detections = set()
+        for i, box in enumerate(boxes):
+            metadata = {
+                "raw_box": boxes[i],
+                "class": self.category_index[classes[i]],
+                "score": scores[i]
+            }
```

Comment:
This would be better:
```diff
diff --git a/stonesoup/detector/tensorflow.py b/stonesoup/detector/tensorflow.py
index 6c5f2db..1db4d68 100644
--- a/stonesoup/detector/tensorflow.py
+++ b/stonesoup/detector/tensorflow.py
@@ -178,11 +178,11 @@ class TensorFlowBoxObjectDetector(Detector):
         # Empty detection set
         frame_height, frame_width, _ = frame.pixels.shape
         detections = set()
-        for i, box in enumerate(boxes):
+        for box, class_, score in zip(boxes, classes, scores):
             metadata = {
-                "raw_box": boxes[i],
-                "class": self.category_index[classes[i]],
-                "score": scores[i]
+                "raw_box": box,
+                "class": self.category_index[class_],
+                "score": score
             }
             # Transform box to be in format (x, y, w, h)
             box_xy = np.array([[box[1]*frame_width],
```

Reply:
Should be fixed now 👍